opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/CanonicalRootValidator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2019, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.configuration;
+
+import java.io.File;
+
+/**
+ * Represents a validator of a --canonicalRoot value.
+ */
+public class CanonicalRootValidator {
+
+    /**
+     * Validates the specified setting, returning an error message if validation
+     * fails.
+     * @return a defined instance if not valid or {@code null} otherwise
```

Comment:
I'd say this sort of return value semantics is a bit unusual however (I'd expect an exception thrown by static class) it is quite clearly marked in the code that uses it (`problem = validate(...)`) so that's fine with me. I think Golang error return style has something to it.

Reply:
Thank you for looking. Yes since it was going to be used in two different manners — once for a fatal termination and another time to just skip over an entry — I just made it return a problem description.

Reply:
Okay. I realize not having to handle an exception makes the code less verbose.